src/dune_lang/pform.mli

```diff
@@ -131,17 +132,37 @@ module Macro : sig
   val to_dyn : t -> Dyn.t
 end
 
+module Macro_invocation : sig
+  type t =
+    { macro : Macro.t
+    ; payload : Payload.t
+    }
+
+  (** Similar to [Payload.Args] but incorporates the macro name into error messages *)
+  module Args : sig
+    (** Treat the entire payload as a single string argument. *)
+    val whole : t -> string
+
+    (** Split the payload on the first ':' character raising a [User_error] if the
+        payload contains no ':' characters. *)
+    val lsplit2_exn : t -> Loc.t -> string * string
```

Comment:
I made this non optional. Omitting locations is almost always a mistake for something like this.